### DISCUSSION TITLE
Nesting variables within image variables

### DISCUSSION THREAD

**Discussion by CaeriTech** [brndnmtthws/conky#1921] [Q&A]
I can't seem to figure out how to nest variables in the image variable.

e.g. ${image /tmp/$USER~myImage~d_3.png -p  61,336 -s 30x30} does not show an image and I see no error messages.  Is this supported?

${execi 2130 grep --extended-regexp 'h_hour1:' /tmp/$USER~openweather} does however produce the text as expected.

**Comment by Caellian** [comment]
> e.g. ${image /tmp/$USERmyImaged_3.png -p 61,336 -s 30x30} does not show an image and I see no error messages. Is this supported?

No, variables can't be nested - with exception of [`templates`](https://conky.cc/variables#templateN). I believe their content is directly substituted into final text so using `template0 = [[$user]]` and then `${image /tmp/${template0}myImaged_3.png -p 61,336 -s 30x30}` sadly won't work.

Recursive variable resolution is a non-trivial problem to solve (right).

**Reply by CaeriTech** [reply]
Thank you for confirming.
